charmcraft/commands/pack.py

```diff
@@ -58,27 +65,93 @@ def get_paths_to_include(config):
 
 
 _overview = """
-Build the bundle and package it as a zip archive.
+Build and pack a charm operator package or a bundle.
 
-You can `juju deploy` the bundle .zip file or upload it to
-the store (see the "upload" command).
+You can `juju deploy` the resulting `.charm` or bundle's `.zip`
+file directly, or upload it to Charmhub with `charmcraft upload`.
+
+For the charm you must be inside a charm directory with a valid
+`metadata.yaml`, `requirements.txt` including the `ops` package
+for the Python operator framework, and an operator entrypoint,
+usually `src/charm.py`.  See `charmcraft init` to create a
+template charm directory structure.
+
+For the bundle you must already have the `bundle.yaml` (can be
+generated by Juju) and the README.md file.
 """
 
 
 class PackCommand(BaseCommand):
     """Build the bundle or the charm.
 
-    Eventually this command will also support charms, but for now it will work only
-    on bundles.
+    If charmcraft.yaml missing or indicates a charm, use the "build"
+    infrastructure to build the charm.
+
+    Otherwise pack the bundle.
     """
 
     name = "pack"
-    help_msg = "Build the bundle"
+    help_msg = "Build the charm or bundle"
     overview = _overview
-    needs_config = True
+    needs_config = False  # optional until we fully support charms here
+
+    def fill_parser(self, parser):
+        """Add own parameters to the general parser."""
+        parser.add_argument(
+            "-e",
+            "--entrypoint",
```

Comment:
Is `-- entrypoint` already agreed upon, or should it be `--entry-point`?

Reply:
I'd stick with the same exact parameter that comes from the "build" command. In the medium term we need to discuss if we should even keep these two options, they probably should be options for the "Charm Plugin". Let's talk about this.

Reply:
This should probably go away and become a charmcraft.yaml proper field equivalent to apps or something the charm plugin deals with